store/src/java/com/zimbra/cs/service/mail/ItemAction.java

```diff
@@ -190,11 +190,12 @@ protected String handleCommon(Map<String, Object> context, Element request, Stri
                 localResults = ItemActionHelper.RENAME(octxt, mbox, responseProto, local, type, tcon, name, iidFolder).getResult();
             } else if (opStr.equals(MailConstants.OP_UPDATE)) {
                 String folderId = action.getAttribute(MailConstants.A_FOLDER, null);
-                ItemId iidFolder = new ItemId(folderId == null ? "-1" : folderId, zsc);
-                if (!iidFolder.belongsTo(mbox)) {
-                    throw ServiceException.INVALID_REQUEST("cannot move item between mailboxes", null);
-                } else if (folderId != null && iidFolder.getId() <= 0) {
```

Comment:
This code was introduced for https://bugzilla.zimbra.com/show_bug.cgi?id=9333 (blame used in zm-store repo to find that)  Commit comment was:

    We were silently ignoring negative target folder IDs in op="update".  Now we throw mail.NO_SUCH_FOLDER.

Need to re-instate code to throw NO_SUCH_FOLDER if a negative folder ID has been supplied.